crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -180,20 +272,919 @@ fn __num_basis_gates(basis_b: f64, basis_fidelity: f64, unitary: MatRef<c64>) ->
+#[allow(non_snake_case)]
+#[pyclass(module = "qiskit._accelerate.two_qubit_decompose", subclass)]
+pub struct TwoQubitWeylDecomposition {
+    #[pyo3(get)]
+    a: f64,
+    #[pyo3(get)]
+    b: f64,
+    #[pyo3(get)]
+    c: f64,
+    #[pyo3(get)]
+    global_phase: f64,
+    K1l: Array2<Complex64>,
+    K2l: Array2<Complex64>,
+    K1r: Array2<Complex64>,
+    K2r: Array2<Complex64>,
+    #[pyo3(get)]
+    specialization: Specializations,
+    default_euler_basis: String,
+    #[pyo3(get)]
+    requested_fidelity: Option<f64>,
+    #[pyo3(get)]
+    calculated_fidelity: f64,
+    unitary_matrix: Array2<Complex64>,
+}
+
+impl TwoQubitWeylDecomposition {
+    fn weyl_gate(
+        &mut self,
+        simplify: bool,
+        sequence: &mut Vec<(String, Vec<f64>, [u8; 2])>,
+        atol: f64,
+        global_phase: &mut f64,
+    ) {
+        match self.specialization {
+            Specializations::MirrorControlledEquiv => {
+                sequence.push(("swap".to_string(), Vec::new(), [0, 1]));
+                sequence.push(("rzz".to_string(), vec![(PI4 - self.c) * 2.], [0, 1]));
+                *global_phase += PI4
+            }
+            Specializations::SWAPEquiv => {
+                sequence.push(("swap".to_string(), Vec::new(), [0, 1]));
+                *global_phase -= 3. * PI / 4.
+            }
+            _ => {
+                if !simplify || self.a.abs() > atol {
+                    sequence.push(("rxx".to_string(), vec![-self.a * 2.], [0, 1]));
+                }
+                if !simplify || self.b.abs() > atol {
+                    sequence.push(("ryy".to_string(), vec![-self.b * 2.], [0, 1]));
+                }
+                if !simplify || self.c.abs() > atol {
+                    sequence.push(("rzz".to_string(), vec![-self.c * 2.], [0, 1]));
+                }
+            }
+        }
+    }
+}
+
+const IPZ: [[Complex64; 2]; 2] = [
+    [C1_IM, Complex64::new(0., 0.)],
+    [Complex64::new(0., 0.), Complex64::new(0., -1.)],
+];
+const IPY: [[Complex64; 2]; 2] = [
+    [Complex64::new(0., 0.), Complex64::new(1., 0.)],
+    [Complex64::new(-1., 0.), Complex64::new(0., 0.)],
+];
+const IPX: [[Complex64; 2]; 2] = [
+    [Complex64::new(0., 0.), C1_IM],
+    [C1_IM, Complex64::new(0., 0.)],
+];
+
+#[pymethods]
+impl TwoQubitWeylDecomposition {
+    fn __getstate__(&self, py: Python) -> ([f64; 5], [PyObject; 5], u8, String, Option<f64>) {
+        let specialization = match self.specialization {
+            Specializations::General => 0,
+            Specializations::IdEquiv => 1,
+            Specializations::SWAPEquiv => 2,
+            Specializations::PartialSWAPEquiv => 3,
+            Specializations::PartialSWAPFlipEquiv => 4,
+            Specializations::ControlledEquiv => 5,
+            Specializations::MirrorControlledEquiv => 6,
+            Specializations::SimaabEquiv => 7,
+            Specializations::SimabbEquiv => 8,
+            Specializations::SimabmbEquiv => 9,
+        };
```

Comment:
Done in: https://github.com/Qiskit/qiskit/pull/11946/commits/5fac9acd72b937b19e5f08fce5305a83682a6890 I didn't use a pythmethod though because the pickling gets a bit odd for enums because they don't really have a constructor.